airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -351,17 +352,73 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
+    ) -> MutableMapping[str, Any]:
         next_page_token = next_page_token or {}
+        if stream_state:
+            # use the state value if exists
+            parsed_state = calendar.timegm(pendulum.parse(stream_state.get(self.cursor_field)).utctimetuple())
+        else:
+            # for full-refresh use start_date
+            parsed_state = calendar.timegm(pendulum.parse(self._start_date).utctimetuple())
         if self.cursor_field:
-            params = {
-                "start_time": next_page_token.get(self.cursor_field, calendar.timegm(pendulum.parse(self._start_date).utctimetuple()))
-            }
+            params = {"start_time": next_page_token.get(self.cursor_field, parsed_state)}
         else:
             params = {"start_time": calendar.timegm(pendulum.parse(self._start_date).utctimetuple())}
         return params
 
 
+class ZendeskSupportTicektEventsExportStream(SourceZendeskSupportCursorPaginationStream):
+    """Incremental Export from TicketEvents stream:
+    https://developer.zendesk.com/api-reference/ticketing/ticket-management/incremental_exports/#incremental-ticket-event-export
+    
+    @ param response_list_name: the main nested entity to look at inside of response, defualt = "ticket_events"
+    @ param responce_target_entity: nested property inside of `response_list_name`, default = "child_events"
+    @ param list_entities_from_event : the list of nested child_events entities to include from parent record
+    @ param sideload_param : parameter variable to include various information to child_events property
+        more info: https://developer.zendesk.com/documentation/ticketing/using-the-zendesk-api/side_loading/#supported-endpoints
+    @ param event_type : specific event_type to check ["Audit", "Change", "Comment", etc]
+    """
+    
+    response_list_name: str = "ticket_events"
+    responce_target_entity: str = "child_events"
+    list_entities_from_event: List[str] = None
+    sideload_param: str = None
+    event_type: str = None
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        Returns next_page_token based on `end_of_stream` parameter inside of response
+        """
+        next_page_token = super().next_page_token(response)
+        end_of_stream = response.json().get(END_OF_STREAM_KEY, False)
+        return None if end_of_stream else next_page_token
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token, **kwargs)
+        if self.sideload_param:
+            params["include"] = self.sideload_param
+        return params
+    
+    def update_event_props(self, record: dict = None, event: dict = None, props: list = None) -> MutableMapping[str, Any]:
```

Comment:
could you add a unit test for all methods which can be unit tested?

Reply:
Will do.